nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -478,6 +488,20 @@ in
           /nix/var/nix/gcroots/tmp
       '';
 
+    nix.systemFeatures = mkIf (pkgs.stdenv.isx86_64) (
+       # can build for arch
+      mkDefault {
+        "default"        = [ ];
+        "westmere"       = [ "gccarch-westmere" ];
+        "sandybridge"    = [ "gccarch-westmere" "gccarch-sandybridge" ];
+        "ivybridge"      = [ "gccarch-westmere" "gccarch-sandybridge" "gccarch-ivybridge" ];
+        "haswell"        = [ "gccarch-westmere" "gccarch-sandybridge" "gccarch-ivybridge" "gccarch-haswell" ];
+        "broadwell"      = [ "gccarch-westmere" "gccarch-sandybridge" "gccarch-ivybridge" "gccarch-haswell" "gccarch-broadwell" ];
+        "skylake"        = [ "gccarch-westmere" "gccarch-sandybridge" "gccarch-ivybridge" "gccarch-haswell" "gccarch-broadwell" "gccarch-skylake" ];
+        "skylake-avx512" = [ "gccarch-westmere" "gccarch-sandybridge" "gccarch-ivybridge" "gccarch-haswell" "gccarch-broadwell" "gccarch-skylake" "gccarch-skylake-avx512" ];
+      }.${pkgs.hostPlatform.platform.gcc.arch or "default"}
```

Comment:
One could see an total order here which temptes to write a comparision function and get rid of those matrixes.
But it is so only because there are no AMD processors yet, only Intel.